src/System.Net.Security/src/System/Net/SecurityStatusAdapterPal.Windows.cs

```diff
@@ -0,0 +1,107 @@
+            new SecurityStatusMapping(Interop.SecurityStatus.LogonDenied, SecurityStatusPal.LogonDenied),
+            new SecurityStatusMapping(Interop.SecurityStatus.NoAuthenticatingAuthority, SecurityStatusPal.NoAuthenticatingAuthority),
+            new SecurityStatusMapping(Interop.SecurityStatus.NoImpersonation, SecurityStatusPal.NoImpersonation),
+            new SecurityStatusMapping(Interop.SecurityStatus.NoCredentials, SecurityStatusPal.NoCredentials),
+            new SecurityStatusMapping(Interop.SecurityStatus.NotOwner, SecurityStatusPal.NotOwner),
+            new SecurityStatusMapping(Interop.SecurityStatus.OK, SecurityStatusPal.OK),
+            new SecurityStatusMapping(Interop.SecurityStatus.OutOfMemory, SecurityStatusPal.OutOfMemory),
+            new SecurityStatusMapping(Interop.SecurityStatus.OutOfSequence, SecurityStatusPal.OutOfSequence),
+            new SecurityStatusMapping(Interop.SecurityStatus.PackageNotFound, SecurityStatusPal.PackageNotFound),
+            new SecurityStatusMapping(Interop.SecurityStatus.MessageAltered, SecurityStatusPal.MessageAltered),
+            new SecurityStatusMapping(Interop.SecurityStatus.NoCredentials, SecurityStatusPal.NoCredentials),
+            new SecurityStatusMapping(Interop.SecurityStatus.QopNotSupported, SecurityStatusPal.QopNotSupported),
+            new SecurityStatusMapping(Interop.SecurityStatus.Renegotiate, SecurityStatusPal.Renegotiate),
+            new SecurityStatusMapping(Interop.SecurityStatus.SecurityQosFailed, SecurityStatusPal.SecurityQosFailed),
+            new SecurityStatusMapping(Interop.SecurityStatus.SmartcardLogonRequired, SecurityStatusPal.SmartcardLogonRequired),
+            new SecurityStatusMapping(Interop.SecurityStatus.TargetUnknown, SecurityStatusPal.TargetUnknown),
+            new SecurityStatusMapping(Interop.SecurityStatus.TimeSkew, SecurityStatusPal.TimeSkew),
+            new SecurityStatusMapping(Interop.SecurityStatus.UnknownCredentials, SecurityStatusPal.UnknownCredentials),
+            new SecurityStatusMapping(Interop.SecurityStatus.UnsupportedPreauth, SecurityStatusPal.UnsupportedPreauth),
+            new SecurityStatusMapping(Interop.SecurityStatus.Unsupported, SecurityStatusPal.Unsupported),
+            new SecurityStatusMapping(Interop.SecurityStatus.UntrustedRoot, SecurityStatusPal.UntrustedRoot),
+            new SecurityStatusMapping(Interop.SecurityStatus.WrongPrincipal, SecurityStatusPal.WrongPrincipal),
+        };
+
+        internal static SecurityStatusPal GetSecurityStatusPalFromWin32Int(int win32SecurityStatus)
+        {
+            return GetSecurityStatusPalFromInterop((Interop.SecurityStatus) win32SecurityStatus);
+        }
+
+        internal static SecurityStatusPal GetSecurityStatusPalFromInterop(Interop.SecurityStatus win32SecurityStatus)
+        {
+            foreach (SecurityStatusMapping mapping in s_securityStatusMapping)
```

Comment:
I just noticed the original was a case statement, which will have performance akin to a dictionary rather than a traversal like this.